projects/plugins/jetpack/modules/subscriptions/jetpack-user-content-link-redirection.php

```diff
@@ -48,6 +69,29 @@ function jetpack_user_content_link_redirection() {
 	exit;
 }
 
+/**
+ * Check if the domain of the request is the same as the domain of the given blog_id
+ *
+ * @param string $domain_from_request The domain from the request.
+ * @param int    $blog_id             The blog_id to check against.
+ *
+ * @return bool
+ */
+function jetpack_user_content_link_is_same_domain( $domain_from_request, $blog_id ) {
+	$request  = sprintf( '/sites/%d?fields=URL', $blog_id );
+	$response = Client::wpcom_json_api_request_as_blog( $request );
+	if ( is_wp_error( $response ) ) {
+		return false;
+	}
+
+	$domain_from_wpcom = wp_parse_url( json_decode( wp_remote_retrieve_body( $response ) )->URL, PHP_URL_HOST );
```

Comment:
Is there a difference in the URL from`get_home_url()` vs. the URL from the WPCOM Sites API endpoint? Otherwise, it wouldn't be necessary to make an API endpoint call just to get a URL.

Reply:
I checked, and the `get_home_url()` behaves differently on Jetpack. It always returns the home for the current site and ignores the blog_id param.

Reply:
So this is for the situation where `blog_id` could be different from the current Jetpack site?

Reply:
Moved to the blog_id check instead to simplify the implementation. It should cover all the scenarios where blog IDs and domains do not match.